backend/ftp/dataconn_test.go

```diff
@@ -19,145 +19,184 @@ import (
 
 type dataConnSuite struct {
 	suite.Suite
+	ftpFile *File
+	client  *mocks.Client
```

Comment:
most changes in this test are simple refactoring of 1 long, single test broken into multiple tests.  Other than that, updating the `List` asserts to match the parent directory "/", rather than the location dir "/some/" and returning  List entries that includes "some" as a directory entry. The refactor makes it look worse than it is.